### PR TITLE
Proposal Maintenance Funcs + Fixes

### LINKED ISSUE #263: add setAuthorizer and setPaymentProcessor functions (and probably setFundingManager)
Right now once the address is set there's no way to change it - even if we remove the module itself from the proposal. This is clearly bad. There should be an onlyAuthorized function to modify it

### LINKED ISSUE #278: Fix pre-commit hook
Currently, the pre-commit hook only works of certain environment variables are loaded since it runs the scripts as well - which is not ideal for a pre-commit hook. We will need to adapt the makefile to include the required env variables (just like with the `make testFuzz` command) so that it works without any prior work.

Currently the output is:
```
Error: 
Failed to deploy script:
Execution reverted: EvmError: Revert (gas: 1500082)
make[1]: *** [testScripts] Error 1
make: *** [pre-commit] Error 2
```

### PR DESCRIPTION
**Pull request by marvinkruse**
This pr introduces:
* Fixes #278 (allowing the pre-commit hook to be ran without sourcing any `.env` file)
* Fixes #263 (adding maintenance funcs to the proposal + tests)

The failing test on mainnet should also be done now.

### CODE REVIEW

**Review by FHieser — APPROVED** [review]

File: src/proposal/Proposal.sol [hunk 1]
```diff
@@ -136,6 +136,33 @@ contract Proposal is IProposal, OwnableUpgradeable, ModuleManager {
     //--------------------------------------------------------------------------
     // onlyAuthorized Functions
 
+    /// @inheritdoc IProposal
+    function setAuthorizer(IAuthorizer authorizer_) external onlyAuthorized {
```
Comment (saxenism) [inline src/proposal/Proposal.sol:140]: Perhaps it'll be cool to emit events when such major changes are taking place. Might come in handy for the FE team?

File: test/proposal/Proposal.t.sol [hunk 2]
```diff
@@ -231,4 +359,23 @@ contract ProposalTest is Test {
...
+    {
+        uint length = addrs.length;
+        vm.assume(length > 0); //Array has to be at least 1
+
+        if (length <= size) {
+            return addrs;
+        }
+
+        address[] memory cutArry = new address[](size);
+        for (uint i = 0; i < size - 1; i++) {
```
Comment (saxenism) [inline test/proposal/Proposal.t.sol]: unchecked pre-increments can make the tests run faster.

Also no need to initialize the `uint i` variable.

File: src/proposal/Proposal.sol [hunk 3]
```diff
@@ -136,6 +136,33 @@ contract Proposal is IProposal, OwnableUpgradeable, ModuleManager {
...
+    }
+
+    /// @inheritdoc IProposal
+    function setFundingManager(IFundingManager fundingManager_)
+        external
+        onlyAuthorized
+    {
+        addModule(address(fundingManager_));
+        removeModule(address(fundingManager));
+        fundingManager = fundingManager_;
```
Comment (saxenism) [inline src/proposal/Proposal.sol:154]: Same. Can we emit an event about the change?

File: src/proposal/Proposal.sol [hunk 4]
```diff
@@ -136,6 +136,33 @@ contract Proposal is IProposal, OwnableUpgradeable, ModuleManager {
...
+    }
+
+    /// @inheritdoc IProposal
+    function setPaymentProcessor(IPaymentProcessor paymentProcessor_)
```
Comment (saxenism) [inline src/proposal/Proposal.sol:159]: Same. Can we emit an event about the change?

File: test/proposal/Proposal.t.sol (hunk 2 above)
Comment (marvinkruse) [reply test/proposal/Proposal.t.sol]: It's in a test, where we have quite some loops that are not optimized, but good point still - @FHieser fixed it 💪

**Pull request merged**